src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -3266,10 +3266,11 @@ private function registerRateLimiterConfiguration(array $config, ContainerBuilde
             $limiterConfig['id'] = $name;
             $limiter->replaceArgument(0, $limiterConfig);
 
-            $container->registerAliasForArgument($limiterId, RateLimiterFactory::class, $name.'.limiter');
+            $factoryAlias = $container->registerAliasForArgument($limiterId, RateLimiterFactory::class, $name.'.limiter');
 
             if (interface_exists(RateLimiterFactoryInterface::class)) {
                 $container->registerAliasForArgument($limiterId, RateLimiterFactoryInterface::class, $name.'.limiter');
+                $factoryAlias->setDeprecated('symfony/dependency-injection', '7.3', 'The "%alias_id%" service alias is deprecated and will be removed in 8.0.').\sprintf(' You should use "%s %s.limiter" instead.', RateLimiterFactoryInterface::class, $name);
```

Comment:
There is something wrong with the concatenation `').\sprintf('` it looks like it’s outside the message.

Reply:
Yeah 🤮, but I just remembered I can escape the `%` in `%alias_id%`. I'll fix this.